syntax.php

```diff
@@ -36,9 +36,16 @@ function handle($match, $state, $pos, Doku_Handler $handler) {
             $type = 'word';
         }
 
-        list($num, $ns) = explode('>', $match, 2);
-        list($junk, $num) = explode(':', $num, 2);
-        $flags = null;
+        // Add trailing separator to ensure we always have 2 entries in the array
+        list($num, $ns) = explode('>', $match . '>', 2);
+        $ns = rtrim($ns, '>');
+        list($junk, $num) = explode(':', $num . ':', 2);
+        $num = rtrim($num, ':');
+
+        // Set default flag values
+        $flags = [
```

Comment:
$flags is also fed to the preg_match, so you should set this later at line 52, not here. The check on line 145 is then not needed anymore.

I prefer the other way, but of course this works for the explode-list combi as well.